packages/web-frontend/cypress/scripts/generateDashboardReportConfig.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import { reduceToDictionary } from '@tupaia/utils';
+
+const getProjectCodeToName = async database => {
+  const projects = await database.executeSql(`
+    SELECT p.code, e.name FROM project p
+    JOIN entity e on e.id = p.entity_id
+  `);
+  return reduceToDictionary(projects, 'code', 'name');
+};
+
+const getDashboardReportIdToGroup = async database => {
+  const dashboardGroups = await database.executeSql(`SELECT * from "dashboardGroup"`);
+
+  const reportIdToGroup = {};
+  dashboardGroups.forEach(dashboardGroup => {
+    dashboardGroup.dashboardReports.forEach(reportId => {
+      reportIdToGroup[reportId] = dashboardGroup;
+    });
+  });
+  return reportIdToGroup;
+};
+
+/**
+ * We generate the least amount of config required to test each report once.
+ * 1. For each report, use the first dashboard group that includes it
+ * 2. For each dashboard group, use the first project that includes it
```

Comment:
Nice!

Reply:
Just to be persnickety, it's not actually "the first" dashboard group, but rather the last based on the order the db returns them

Reply:
Oh good point! By the way, we may want to change this strategy to include more combinations, or we can support multiple strategies based on a script option.... But for now the "minimum" combinations strategy will work